### PR TITLE
Redirect when on home so we can see architecture properly

### LINKED ISSUE #22: Architecture image is broken in the landing page
When visiting [this link](https://tikv.github.io/deep-dive-tikv/), we are visiting the first section (i.e. Introduction). But the architecture image inside is broken.

### PR DESCRIPTION
**Pull request by Hoverbear**
Fixes #22.

We do this by redirection. 🤷‍♀️ There doesn't really seem to be a better way.

### CODE REVIEW

File: theme/header.hbs [hunk 1]
```diff
@@ -6,3 +6,10 @@
 <!-- To automatically render math in text elements, include the auto-render extension: -->
 <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js" integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF" crossorigin="anonymous"
     onload="renderMathInElement(document.body);"></script>
+
+<!-- Redirect to the overview page if on home, otherwise the image won't work on one of the following: local index, prod index, prod chapter -->
+<script defer>
```
Comment (breezewish) [inline theme/header.hbs]: - looks like the `defer` attribute is not needed
- should this `<script>` be placed before other script tags?
- what's the purpose of matching path `/deep-dive-tikv/`?

Comment (Hoverbear) [reply theme/header.hbs]: We only want it to cover the URL https://tikv.github.io/deep-dive-tikv/, and we can't directly edit the index.html. Though from your comment I've realized I can localize this just to the particular section.. 😁 

I'll make a change, thanks!

File: theme/header.hbs [hunk 2]
```diff
@@ -5,4 +5,4 @@
 
 <!-- To automatically render math in text elements, include the auto-render extension: -->
 <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-rc.1/dist/contrib/auto-render.min.js" integrity="sha384-yACMu8JWxKzSp/C1YV86pzGiQ/l1YUfE8oPuahJQxzehAjEt2GiQuy/BIvl9KyeF" crossorigin="anonymous"
-    onload="renderMathInElement(document.body);"></script>
+    onload="renderMathInElement(document.body);"></script>
```
Comment (breezewish) [inline theme/header.hbs]: You may want to keep the new line ending :)

File: src/overview/introduction.md [hunk 3]
```diff
@@ -35,3 +35,10 @@ All the network communications are through [gRPC](https://grpc.io/) so that cont
 The whole cluster is managed and scheduled by a central service &mdash; [Placement Driver(PD)](https://github.com/pingcap/pd).
 
 As you can see, the hierarchy of TiKV is clear and easy to understand, and we will give more detailed explanation later.
+
+<!-- Redirect to the overview page if on home, otherwise the image won't work on one of the following: local index, prod index, prod chapter -->
+<script>
+    if (window.location.pathname == "/deep-dive-tikv/" || window.location.pathname == "/") {
+        window.location.replace("overview/introduction.html");
```
Comment (breezewish) [inline src/overview/introduction.md:42]: Is it better to use `/deep-dive-tikv/overview/introduction.html`?

File: src/overview/introduction.md [hunk 4]
```diff
@@ -35,3 +35,10 @@ All the network communications are through [gRPC](https://grpc.io/) so that cont
 The whole cluster is managed and scheduled by a central service &mdash; [Placement Driver(PD)](https://github.com/pingcap/pd).
 
 As you can see, the hierarchy of TiKV is clear and easy to understand, and we will give more detailed explanation later.
+
+<!-- Redirect to the overview page if on home, otherwise the image won't work on one of the following: local index, prod index, prod chapter -->
+<script>
+    if (window.location.pathname == "/deep-dive-tikv/" || window.location.pathname == "/") {
```
Comment (breezewish) [inline src/overview/introduction.md:41]: When will `/` being matched?

**Review by breezewish — COMMENTED** [review]
Mostly fine! I'm only not sure about what kind of URL you want to match. I can't find somewhere that pathname is `/`.

File: src/overview/introduction.md (hunk 4 above)
Comment (Hoverbear) [reply src/overview/introduction.md:41]: Localhost serve from `mdbook serve` to replicate the behavior!

File: src/overview/introduction.md (hunk 3 above)
Comment (Hoverbear) [reply src/overview/introduction.md:42]: No because on localhost serving this is an invalid URL.

File: theme/header.hbs (hunk 2 above)
Comment (Hoverbear) [reply theme/header.hbs]: I'll do that, thanks.

**Review by breezewish — APPROVED** [review]

**Pull request merged**